### PR TITLE
Parsing for episodes

### PR DESCRIPTION
**Pull request by daleyp24**
opening for visibility 

## JIRA Ticket
-  https://ihm-it.atlassian.net/browse/IHRACP-4401

## Description
- Creating a no henson episode ingestion service. Updating to_dict method for conversion and comparison in episode ingestion

#### Major Changes:
Dict method updates for comparing against database data and change the object struct

###Related PRS
https://github.com/iheartradio/episodeingestion/pull/1
https://github.com/iheartradio/pyPodcastParser/pull/9
https://github.com/iheartradio/ingestion-talk/pull/148
https://github.com/iheartradio/podcast-utils/pull/31

Modified: (config files not tracked by git)
ingestion deployment 
ingestion talk
discern.

### CODE REVIEW

**Comment by daleyp24** [comment]
TODO: Update uts

**Comment by daleyp24** [comment]
File "/Users/kellymcgaw/.pyenv/versions/3.9.11/envs/revise/lib/python3.9/site-packages/pypodcastparser/Item.py", line 173, in set_description
    self.description =  p.sub('', tag.string)
TypeError: expected string or bytes-like object

issue in revise with 1.4.0

File: pypodcastparser/Item.py [hunk 1]
```diff
@@ -171,10 +165,12 @@ def set_author(self, tag):
     def set_description(self, tag):
         """Parses description and set value."""
         try:
+            p = re.compile(r'<.*?>')
             if (self.content_encoded is not None):
-                self.description = self.content_encoded
+                #Strip html
+                self.description = p.sub('', self.content_encoded)
```
Comment (daleyp24) [inline pypodcastparser/Item.py]: remove per nate

Comment (daleyp24) [reply pypodcastparser/Item.py]: done

File: pypodcastparser/Podcast.py [hunk 2]
```diff
@@ -357,8 +357,13 @@ def set_published_date(self, tag):
         try:
             self.published_date = tag.string
             self.published_date_string = tag.string
-            pubdate = datetime.datetime.strptime(self.published_date, "%a, %d %b %Y %H:%M:%S %Z")
-            self.published_date = datetime.datetime.strftime(pubdate, "%Y-%d-%m, %H:%M:%S")
+            a = tag.string.split(":")
+            b = a[2]
```
Comment (KMichaelMcGaw) [inline pypodcastparser/Podcast.py]: Can we employ more accurate variable-naming, and maybe a comment explaining why this is necessary?

**Review by DrewJoseyGit — COMMENTED** [review]
Maybe we store the numbers such as 32767, -32767 and possibly 3600, 60 as constants.  That way we get a little more info on what they are from the name, and it makes it easier to update them later if need be.

File: pypodcastparser/Item.py [hunk 3]
```diff
@@ -236,13 +232,17 @@ def set_itunes_episode(self, tag):
         """Parses the episode number and sets value"""
         try:
             self.itunes_episode = tag.string
+            if –32,767 > self.itunes_episode or self.itunes_episode > 32,767 or self.itunes_episode == None:
```
Comment (KMichaelMcGaw) [inline pypodcastparser/Item.py]: I'd suggest having the smallint-checker be in `episodeingestion`, that way we can log the offending episodes guid, show_id, and whatever column that isn't in range before we set the value to 0.

**Review by KMichaelMcGaw — CHANGES_REQUESTED** [review]

File: pypodcastparser/Item.py [hunk 4]
```diff
@@ -236,23 +232,27 @@ def set_itunes_episode(self, tag):
...
+            self.itunes_season = 0
 
     def set_itunes_episode_type(self, tag):
         """Parses the episode type and sets value"""
         try:
             self.itunes_episode_type = tag.string
             self.itunes_episode_type = self.itunes_episode_type.lower()
         except AttributeError:
-            self.itunes_episode_type = None
+            self.itunes_episode_type = 0
```
Comment (KMichaelMcGaw) [inline pypodcastparser/Item.py]: We may need to be a little cautious with what we set it to on error, as setting it to 0 will cause it to be rejected by the DB.  The database only accepts `trailer`, `full`, or `bonus` per the DDL.

```    episode_type        text
        constraint episode_episode_type_check
            check (episode_type = ANY (ARRAY ['full'::text, 'trailer'::text, 'bonus'::text])),```

Comment (daleyp24) [reply pypodcastparser/Item.py]: Yep, that was a mistake on my part.

File: pypodcastparser/Item.py (hunk 3 above)
Comment (daleyp24) [reply pypodcastparser/Item.py]: Do we care about that information? What would we even do with that info?

File: pypodcastparser/Podcast.py (hunk 2 above)
Comment (daleyp24) [reply pypodcastparser/Podcast.py]: good point, will address

**Review by IanNoyes — APPROVED** [review]

File: pypodcastparser/Item.py [hunk 5]
```diff
@@ -207,16 +201,57 @@ def set_guid(self, tag):
...
+
+            regex_array = ["^[a-zA-Z]{3},$","^\d{1,2}$","^[a-zA-Z]{3}$","^\d{4}$","^\d\d:\d\d"]
+            new_array = []
+
+
+            for i,v in enumerate(deconstructed_date):
+                if(re.match(regex_array[i],v)):
+                    new_array.append(v)
+                else:
+                    for x,z in enumerate(deconstructed_date):
```
Comment (KMichaelMcGaw) [inline pypodcastparser/Item.py]: Not to be too much of a stickler but can we have better variable names other than "x,z"

File: tests/test_pyPodcastParser.py [hunk 6]
```diff
@@ -474,6 +474,12 @@ def setUp(self):
         basic_podcast_file = open(basic_podcast_path, "rb")
         self.basic_podcast = basic_podcast_file.read()
         self.podcast = Podcast.Podcast(self.basic_podcast)
+# changing
```
Comment (KMichaelMcGaw) [inline tests/test_pyPodcastParser.py]: Does this need to be removed?

**Review by DrewJoseyGit — APPROVED** [review]

**Comment by daleyp24** [comment]
Major Fields to verify after updates: 
explicit 
duration 
pu date

File: tests/test_pyPodcastParser.py (hunk 6 above)
Comment (daleyp24) [reply tests/test_pyPodcastParser.py]: updated in poonams pr

File: pypodcastparser/Item.py (hunk 5 above)
Comment (daleyp24) [reply pypodcastparser/Item.py]: updated in poonams pr

**Comment by daleyp24** [comment]
> Maybe we store the numbers such as 32767, -32767 and possibly 3600, 60 as constants. That way we get a little more info on what they are from the name, and it makes it easier to update them later if need be.

logic was exported to episode ingestion

File: pypodcastparser/Podcast.py [hunk 7]
```diff
@@ -357,8 +357,13 @@ def set_published_date(self, tag):
         try:
             self.published_date = tag.string
             self.published_date_string = tag.string
-            pubdate = datetime.datetime.strptime(self.published_date, "%a, %d %b %Y %H:%M:%S %Z")
-            self.published_date = datetime.datetime.strftime(pubdate, "%Y-%d-%m, %H:%M:%S")
+            a = tag.string.split(":")
```
Comment (KMichaelMcGaw) [inline pypodcastparser/Podcast.py]: would you mind changing these variable names to something that conveys more information?

**Review by KMichaelMcGaw — CHANGES_REQUESTED** [review]

**Review by DrewJoseyGit — APPROVED** [review]

File: pypodcastparser/Item.py [hunk 8]
```diff
@@ -208,15 +226,74 @@ def set_guid(self, tag):
...
+                    new_array.append(array_value)
+                else:
+                    for inner_index,inner_value in enumerate(deconstructed_date):
+                        if(re.match(regex_array[array_index],inner_value)):
+                            new_array.append(inner_value)
+                            break
+            date_string = new_array[0]+" "+new_array[1]+" "+ new_array[2]+" "+new_array[3]+" "+new_array[4]
+
+            if(len(new_array) != 5):
+                raise AttributeError
```
Comment (IanNoyes) [inline pypodcastparser/Item.py]: Can we add a comment onto this AttributeError?  Maybe something like "length of date array is incorrect expected 5 but got x"

File: setup.py [hunk 9]
```diff
@@ -11,7 +11,7 @@
 setup(
     name='pypodcastparser-ihr',
 
-    version='1.3.0',
+    version='1.4.3-dev',
```
Comment (IanNoyes) [inline setup.py]: Change this before me merge.

**Review by IanNoyes — CHANGES_REQUESTED** [review]
Couple small comments.

File: setup.py (hunk 9 above)
Comment (KMichaelMcGaw) [reply setup.py]: 👍

Comment (daleyp24) [reply setup.py]: upgraded to 1.5.0. will need to in ep ingestion too

File: pypodcastparser/Item.py (hunk 8 above)
Comment (daleyp24) [reply pypodcastparser/Item.py]: added error message

**Review by KMichaelMcGaw — APPROVED** [review]
Excellent work with all the date management stuff!

**Pull request merged**